src/main/kotlin/com/goodchoice/domain/auth/service/AuthService.kt

```diff
@@ -43,11 +46,31 @@ class AuthServiceImpl(
     override fun getCredentialsByEmailOrNull(email: Email): AuthWithCredentials? =
         authRepo.getCredentialsByEmailOrNull(email)
 
+    @Transactional(readOnly = true)
+    override fun getByCredentialsOrNull(email: Email, password: RawPassword): Auth? =
+        authRepo.getCredentialsByEmailOrNull(email)?.let {
+            if (passwordEncoder.matches(password.value, it.passwordHash)) {
+                it
+            } else {
+                null
+            }
```

Comment:
You can use `... ?.takeIf { passwordEncoder.matches(password.value, it.passwordHash) } ?: throw ...`

Reply:
done